ietf/draft-ietf-jmap-calendars-18.txt

```diff
@@ -0,0 +1,3360 @@
+      identity within an account.  The default identity should be used
+      by clients whenever they need to choose an identity for the user
+      within this account, and they do not have any other information on
+      which to make a choice.  For example, if creating a scheduled
+      event in this account, the default identity may be automatically
+      added as an owner.  (But the client may ignore this if, for
+      example, it has its own feature to allow users to choose which
+      identity to use based on the invitees.)
+
+
+
+
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 11]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   A participant in an event corresponds to a ParticipantIdentity if the
+   calendarAddress property of the participant is equivalent to the
+   calendarAddress property of the identity after syntax-based
+   normalisation, as per Section 6.2.2 of [RFC3986].
+
+   The following JMAP methods are supported.
+
+3.1.  ParticipantIdentity/get
+
+   This is a standard "/get" method as described in Section 5.1 of
+   [RFC8620].  The "ids" argument may be null to fetch all at once.
+
+3.2.  ParticipantIdentity/changes
+
+   This is a standard "/changes" method as described in Section 5.2 of
+   [RFC8620].
+
+3.3.  ParticipantIdentity/set
+
+   This is a standard "/set" method as described in Section 5.3 of
+   [RFC8620], but with the following additional request argument:
+
+   *  *onSuccessSetIsDefault*: Id|null
+      If an id is given, and all creates, updates and destroys (if any)
+      succeed without error, the server will try to set this identity as
+      the default.  (For references to ParticipantIdentity creations,
+      this is equivalent to a creation-reference, so the id will be the
+      creation id prefixed with a "#".)
+
+   If the id is not found, or the change is not permitted by the server
+   for policy reasons, it MUST be ignored and the currently default
+   ParticipantIdentity (if any) will remain as such.  No error is
+   returned to the client in this case.
+
+   As per Section 5.3 of [RFC8620], if the default is successfully
+   changed, any changed objects MUST be reported in either the "created"
+   or "updated" argument in the response as appropriate, with the
+   server-set value included.
+
+   The server MAY restrict the URI values the user may claim, for
+   example only allowing mailto: URIs with email addresses that belong
+   to the user.  A standard forbidden error is returned to reject non-
+   permissible changes.
+
+4.  Calendars
+
+   A Calendar is a named collection of events.  All events are
+   associated with at least one calendar.
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 12]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   A *Calendar* object has the following properties:
+
```

Comment:
```   *  *name*: String

      The user-visible name of the calendar.  This may be any UTF-8
      string of at least 1 character in length and maximum 255 octets in
      size.```

The mixture of "character" and "octets" almost seems significant as if character suggests a unicode character which might be more than 1 octet? It feels like a distraction.

For comparison, the jmap contact rfc in last call (https://www.ietf.org/archive/id/draft-ietf-jmap-contacts-10.txt) says:
```

   *  *name*: String

      The user-visible name of the AddressBook.  This MUST NOT be the
      empty string and MUST NOT be greater than 255 octets in size when
      encoded as UTF-8.
```

which is maybe better